chiselFrontend/src/main/scala/chisel3/core/Module.scala

```diff
@@ -101,17 +101,11 @@ extends HasId {
   final val name = Builder.globalNamespace.name(desiredName)
 
   /** FIRRTL Module name */
-  private var _modName: Option[String] = None
-  private[chisel3] def setModName(name: String) = _modName = Some(name)
-  def modName = _modName match {
-    case Some(name) => name
-    case None => throwException("modName should be called after circuit elaboration")
-  }
+  def modName = name
```

Comment:
Can we just delete modName?  It's not supposed to be part of the public API, anyway.